src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -538,7 +539,14 @@ async def create_pex_from_targets(
         sources_digests.append(request.additional_sources)
     if request.include_source_files:
         transitive_targets = await Get(
-            TransitiveTargets, TransitiveTargetsRequest(request.addresses)
+            TransitiveTargets,
+            TransitiveTargetsRequest(
+                request.addresses,
+                should_traverse_deps_predicate=TraverseIfNotPackageTarget(
+                    roots=request.addresses,
+                    union_membership=union_membership,
+                ),
+            ),
         )
         sources = await Get(PythonSourceFiles, PythonSourceFilesRequest(transitive_targets.closure))
 
```

Comment:
Should we also do the right thing and try and `package` the package targets for inclusion?

Reply:
then it's not sources but local dists? which is handled further down, iiuc..

Reply:
Oh that's even weirder I guess. I'm thinking of other packageables, like a `pex_binary` or an `archive` or some plugin-provided packageable thing.

Reply:
are you saying to embed other PEXes/archives/etc... inside PEXes?

Reply:
Sure why not? If someone has declared that dependency (between a Python source and a `pex_binary`) the intent is pretty clear to me... that code depends on that PEX.